crates/wasmtime/src/module.rs

```diff
@@ -925,6 +930,20 @@ impl Module {
     }
 }
 
+#[cfg(feature = "memfd-allocator")]
+impl Drop for ModuleInner {
+    fn drop(&mut self) {
+        // Try to unregister from the MemFdRegistry, if we can. Any
+        // entries in the registry holding memfd images for this
+        // module will have been created lazily, on first
+        // instantiation, so they may not exist at all. However, if
+        // they do, we want to ensure that we do not hold them beyond
+        // when this module is dropped, because they will be useless
+        // afterward.
+        self.engine.memfd_registry().remove(self.module.unique_id());
```

Comment:
The comment here about "Try to unregister" feels a bit misleading because this seems like there's less of an attempt here and more of a "this had better be gone" sort of feeling.

Additionally I'm not sure now that I think about it that this is the right place for this destructor. And actually now that I think for a moment I'm a bit confused about this design.

This design currently has a "registry" in an engine of memfds-per-module, but it seems like an alternative (and perhaps simpler?) design would be to store memfd information here in the `Module` itself. I see now that during the first instantiation of a module we create the memfd image if memfd is enabled, but it seems like that work would be best done at module-creation-time if memfd is enabled.

WIth per-module memfd information I think that would remove the need for a registry entirely?

Reply:
One other benefit of moving information out of the `Engine` is that it would avoid bouncing on a lock every time an instantiation is made which can be somewhat expensive

Reply:
Good point -- there actually was no need for a registry system at all. I've made the `wasmtime::Module` own the memfd images. Thanks!